### PR TITLE
Change PhyloPic date range & schedule interval

### LINKED ISSUE #410: Change Phylopic to @weekly
## Problem

Phylopic is scheduled to run `@daily`, but when we _do_ have data, we typically upsert <10 images.

## Description

We should update the DAG so that it runs `@weekly` instead. This will involve changing the date logic, because I believe that it's set up to only do daily ranges (not weekly ones).

## Alternatives

Leave as-is and just recognize that we're going to have unnecessarily high resolution on the Phylopic images.

## Additional context

Related to #380

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #410 by @AetherUnbound

## Description

This PR makes a number of changes to the PhyloPic provider script to allow it to process records over the course of a week.

The previous implementation would specify a _start date_, but not an _end date_. This meant that, if a DAG for a given day were re-run, it would process all data since that date until now (rather than all data updated on that day).

This PR now uses the [from **and** to options available in the PhyloPic API](http://phylopic.org/api/#method-image-time-range) in order to construct an immutable date range. This will ensure that re-runs of a particular DAG will always process data for the same date range.

I've also made some changes to support moving this from an `@daily` to an `@weekly` schedule interval. The processed date range will be 7 days by default, although that value can be modified in the script or via CLI arguments if running locally. Since the interval _start date_ is passed to script by the DAG factory, we just need to add the number of days onto that date.

I improved some of the wording & tests as well (e.g. `args` were used when `kwargs` was more accurate).

## Testing Instructions

1. `just test`
1. (Optional) Run `python openverse_catalog/dags/providers/provider_api_scripts/phylopic.py --date-start 2022-03-10` and observe that the CLI begins processing records from `2022-03-10` to `2022-03-17` (it can be stopped immediately after this process starts)
1. (Optional) Run `python openverse_catalog/dags/providers/provider_api_scripts/phylopic.py --date-start 2022-03-10 --date-end 2022-03-12` and observe that the CLI only processes 2 days worth of data and exits with the message `No content available`!
1. (Optional) Run the PhyloPic DAG and observe that it covers a week's worth of data.

**NOTE**: You may need to add the DAGs folder to your `PYTHONPATH` in order to run those scripts, i.e. `export PYTHONPATH=$PYTHONPATH:$(pwd)/openverse_catalog/dags` from the repo root.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: tests/dags/providers/provider_api_scripts/test_phylopic.py [hunk 1]
```diff
@@ -31,16 +33,29 @@ def test_get_total_images_correct():
...
+        ({"offset": 0}, "http://phylopic.org/api/a/image/list/0/5"),
+        # Missing/None parameters
+        invalid_endpoint({}, None),
+        invalid_endpoint({"offset": None}, None),
+        invalid_endpoint({"date_start": None}, None),
+        invalid_endpoint({"date_start": None, "date_end": None}, None),
+        invalid_endpoint({"date_start": "2020-02-10", "date_end": None}, None),
+    ],
+)
+def test_create_endpoint_for_IDs(data, expected):
```
Comment (stacimc) [inline tests/dags/providers/provider_api_scripts/test_phylopic.py:56]: ✨

File: openverse_catalog/dags/providers/provider_api_scripts/phylopic.py [hunk 2]
```diff
@@ -255,14 +270,28 @@ def _get_image_info(result, _uuid):
...
+
+
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="PhyloPic API Job", add_help=True)
     parser.add_argument(
-        "--date",
+        "--date-start",
         default="all",
-        help="Identify all images" " from a particular date (YYYY-MM-DD).",
+        help="Identify all images from a particular date (YYYY-MM-DD).",
```
Comment (stacimc) [inline openverse_catalog/dags/providers/provider_api_scripts/phylopic.py:287]: Tiny nit, `from a particular date` reads "images _on that particular date_" to me. Maybe `starting from a particular date`?

File: openverse_catalog/dags/providers/provider_api_scripts/phylopic.py [hunk 3]
```diff
@@ -111,18 +120,24 @@ def _get_total_images():
...
+def _create_endpoint_for_IDs(**kwargs):
     limit = LIMIT
 
-    if args.get("date"):
-        # Get a list of objects uploaded/updated on a given date.
-        date = args["date"]
-        endpoint = f"http://phylopic.org/api/a/image/list/modified/{date}"
+    if (date_start := kwargs.get("date_start")) is not None and (
+        date_end := kwargs.get("date_end")
+    ) is not None:
```
Comment (stacimc) [inline openverse_catalog/dags/providers/provider_api_scripts/phylopic.py:128]: I know this file is linted so definitely feel free to ignore this, but this formatting looks strange to me -- at a glance I read the final paren on L128 as closing the initial paren on L126 😕

**Review by stacimc — APPROVED** [review]
LGTM! Tested great, and excellent test coverage 😄 The new schedule sounds good 👍

File: openverse_catalog/dags/providers/provider_api_scripts/phylopic.py (hunk 2 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/providers/provider_api_scripts/phylopic.py:287]: Oh, I should definitely make this help string better! Thanks for the nudge, I was initially just correcting the spacing on it.

File: openverse_catalog/dags/providers/provider_api_scripts/phylopic.py (hunk 3 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/providers/provider_api_scripts/phylopic.py:128]: Yea...`black` is not being super helpful here :disappointed: I'll see if I can move things around to make it a little clearer.

File: openverse_catalog/dags/providers/provider_api_scripts/phylopic.py (hunk 2 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/providers/provider_api_scripts/phylopic.py]: image
That's a bit clearer! :smile:

File: openverse_catalog/dags/providers/provider_api_scripts/phylopic.py (hunk 3 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/providers/provider_api_scripts/phylopic.py]: Done in [3d5e27a](https://github.com/WordPress/openverse-catalog/pull/423/commits/3d5e27aa5211291e26f09fb3af65ede1ca2d9fc4)!

Comment (zackkrida) [reply openverse_catalog/dags/providers/provider_api_scripts/phylopic.py]: nice fix, that was really strange to try and read!

**Review by zackkrida — APPROVED** [review]
Great work, lgtm.

**Pull request merged**